### PR TITLE
Add support for enacting V2 release scheduled changes

### PR DESCRIPTION
**Pull request by mozbhearsum**
This one should be pretty straightforward. This is built on top of #1265, so most of the changesets shown are not relevant to this PR. The second to last changeset is just a refactor of existing Agent code (the diff is not great, but the large block of code is really just being moved -- no changes to it). The final one is a very small addition to query and enact changes for v2 releases.

### CODE REVIEW

File: agent/src/balrogagent/cmd.py [hunk 1]
```diff
@@ -7,53 +7,68 @@
...
+                logging.debug("Change %s is not ready", change["sc_id"])
+
+
+async def process_v2_scheduled_changes(loop, balrog_api_root, auth0_secrets):
+    for endpoint in V2_SCHEDULED_CHANGE_ENDPOINTS:
+        resp = await client.request(balrog_api_root, endpoint, loop=loop, auth0_secrets=auth0_secrets)
+        for r in filter(lambda r: len(r["scheduled_changes"]) > 0, resp["releases"]):
+            t = time.time()
+            if all([time_is_ready(sc, t) for sc in r["scheduled_changes"]]):
+                await client.request(balrog_api_root, f"{endpoint}/{r['name']}/enact", method="POST", auth0_secrets=auth0_secrets, loop=loop)
```
Comment (nthomas-mozilla) [inline agent/src/balrogagent/cmd.py:63]: I'd suggest adding some logging to this function so it's similar to the v1 code branch, unless that's more verbose than is truly useful.

**Comment by mozbhearsum** [comment]
@nthomas-mozilla The most recent commit is the only thing changed here -- which includes logging, verification of signoffs before trying to enact (oops, big oversight...), and tests(!).

File: agent/tests/test_cmd.py [hunk 2]
```diff
@@ -256,3 +258,487 @@ async def testRightEnactOrderForMultipleEndpointsAtOnce(self, time, time_is_read
...
+            }
+        }
+    )
+    monkeypatch.setattr(balrogagent.cmd.client, "request", fr)
+    time_is_ready = MagicMock(return_value=True)
+    monkeypatch.setattr(balrogagent.cmd, "time_is_ready", time_is_ready)
+    await balrogagent.cmd.run_agent(asyncio.get_event_loop(), "http://balrog.fake", "telemetry", auth0_secrets={}, once=True, raise_exceptions=True)
+
+    # Once per scheduled change
+    assert time_is_ready.call_count == 3
```
Comment (nthomas-mozilla) [inline agent/tests/test_cmd.py:742]: Worth checking the call count for verify_signoff here ?

One case you don't cover here is signoff being required and is present.

File: agent/tests/test_cmd.py [hunk 3]
```diff
@@ -256,3 +258,487 @@ async def testRightEnactOrderForMultipleEndpointsAtOnce(self, time, time_is_read
...
+    )
+    monkeypatch.setattr(balrogagent.cmd.client, "request", fr)
+    time_is_ready = MagicMock(side_effect=[True, True, True, True, False])
+    monkeypatch.setattr(balrogagent.cmd, "time_is_ready", time_is_ready)
+    await balrogagent.cmd.run_agent(asyncio.get_event_loop(), "http://balrog.fake", "telemetry", auth0_secrets={}, once=True, raise_exceptions=True)
+
+    # Once per scheduled change
+    assert time_is_ready.call_count == 5
+    # Once for each v1 endpoint, once for each v2 endpoint, once to enact the one release that was ready
+    assert fr.call_count == 8
```
[the remark this answers is not in the document]
Comment (mozbhearsum) [reply agent/tests/test_cmd.py:675]: Good point.

**Comment by mozbhearsum** [comment]
> Oops, I missed a few things there too. A few comments, and there are some warnings like Unknown pytest.mark.asyncio - is this a typo? in the taskcluster log to check out.

Not a typo...I just forgot to include the new dependency in this patch, whoops.

**Comment by mozbhearsum** [comment]
@nthomas-mozilla - I think you intended me to go ahead and merge this after fixing the things noted above. Unless I hear otherwise, I'm going to go ahead and do that next week.

**Comment by nthomas-mozilla** [comment]
Yes, go right ahead.

**Pull request merged**